homeassistant/components/sensor/rfxtrx.py

```diff
@@ -46,13 +47,16 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
                 if data_type in event.values:
                     data_types = [data_type]
                     break
+        add_devices_callback(sensors)
         for _data_type in data_types:
-            new_sensor = RfxtrxSensor(None, entity_info[ATTR_NAME],
-                                      _data_type, entity_info[ATTR_FIREEVENT])
-            sensors.append(new_sensor)
+            entity_id = generate_entity_id(ENTITY_ID_FORMAT,
+                                           entity_info[ATTR_NAME],
+                                           hass=hass)
+            new_sensor = RfxtrxSensor(None, entity_info[ATTR_NAME], _data_type,
+                                      entity_info[ATTR_FIREEVENT], entity_id)
             sub_sensors[_data_type] = new_sensor
+            add_devices_callback([new_sensor])
```

Comment:
This problem was fixed in [0.32.2](https://home-assistant.io/blog/2016/11/05/hacktoberfest-influxdb-weather/#release-0322---november-7):

 - Use sequential updates for non-async entities to prevent race conditions (@pvizeli)